src/fvm_multicell.hpp

```diff
@@ -170,12 +171,24 @@ class fvm_multicell {
     }
 
     /// return reference to the stimuli
-    stimulus_store_type& stimuli() {
-        return stimuli_;
+    util::optional<mechanism&> find_mechanism(const std::string& name) {
```

Comment:
From discussions earlier, I think it's best not to have this considered as part of the 'normal' API for a solver, but can certainly see the utility as a testing interface.

If you agree, I think it would be good to comment the method to indicate that it is exposing mechanims for testing purposes. Also, if we can get away with just the `const` version below, that would be even safer.